src/main/java/group1/mips_simulator/components/cpuParts/RegisterFile.java

```diff
@@ -1,66 +1,103 @@
 package group1.mips_simulator.components.cpuParts;
+
+import group1.mips_simulator.Utility;
 import group1.mips_simulator.components.Value;
 
 public class RegisterFile {
+
+    public static final int GPR_COUNT = 4;
+    public static final int IXR_COUNT = 3;
+
     /*
     A register file is a collection of all registers accessible and usable by a CPU
 
     Design note: when hardware is taped out and manufactured, there's no way to add/remove registers.
     that's why im replacing the Vector (used in MipsComputer, which im renaming to Computer) with a preset array
      */
-    Register[] GPR = new Register[4];   //General Purpose Reg: R0-R3
-    Register[] IXR = new Register[3];   //Index Reg: X1-X3.
+    Register[] GPR = new Register[GPR_COUNT];   //General Purpose Reg: R0-R3
+    Register[] IXR = new Register[IXR_COUNT];   //Index Reg: X1-X3.
     Register PC = new Register();       //Prog Counter Reg : address of next instruction to be executed
     Register CC = new Register();       //Condition Code Reg : set when arith/logical operations executed
-                                        // VNZE (oVerflow, uNderflow, divZero, Equalornot)
-                                // (only one i dont understand is "equal or not", is that saying, set flat to 1 if equal
-                                // and set to 0 if not?)
+    // VNZE (oVerflow, uNderflow, divZero, Equalornot)
+    // (only one I dont understand is "equal or not", is that saying, set flat to 1 if equal
+    // and set to 0 if not?)
     Register IR = new Register();       //Instruction Reg : holds instruction to be exe
     Register MAR = new Register();      //Mem Addr Reg : holds address of word to be fetched from memory
     Register MBR = new Register();      //Mem Buffer Reg : holds word just-fetched-from/to-be-writen-to memory
     Register MFR = new Register();      //Mem Fault Reg : contains ID code if machine fault after it occurs
     //Register[] FPR = new Register[2]; //float point register placeholder <<DO NOT IMPLEMENT TILL PART 4>>
     //Other registers add here as we need them (prof said he hasnt given us all of them)
 
-    public RegisterFile(){
+    public RegisterFile() {
```

Comment:
its beautiful, nice

Reply:
not a fan of nullValue tho? haha

Reply:
I've got an auto linter installed so every time I save it'll mess with the whitespace. Also, I was getting null exceptions so I decided to fix this so my tests actually worked